packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -716,18 +718,36 @@ export class StripeHelper {
       };
 
       try {
-        const invoice = await this.previewInvoice({
-          country,
-          priceId,
-          promotionCode,
-        });
+        const { currency, discount, total, total_discount_amounts } =
+          await this.previewInvoice({
+            country,
+            priceId,
+            promotionCode,
+          });
+
+        if (discount && total_discount_amounts) {
+          couponDetails.discountAmount = total_discount_amounts[0].amount;
+        }
 
-        if (invoice?.discount && invoice?.total_discount_amounts) {
-          couponDetails.discountAmount =
-            invoice.total_discount_amounts[0].amount;
+        const minAmount = getMinimumAmount(currency);
+        if (total !== 0 && minAmount && total < minAmount) {
+          throw error.invalidPromoCode(promotionCode);
+        }
+      } catch (error) {
+        if (
+          error instanceof AppError &&
+          error.errno === AppError.ERRNO.INVALID_PROMOTION_CODE
+        ) {
+          throw error;
+        } else {
+          Sentry.withScope((scope) => {
+            scope.setContext('retrieveCouponDetails', {
+              priceId,
+              promotionCode,
+            });
+            Sentry.captureException(error);
+          });
```

Comment:
I cannot find a test for this new code.

Reply:
Updated the one test to check that Sentry was called, and added another one for getMinAmount exceptions.